### ISSUE TITLE
Fix macro labels

### ISSUE THREAD

**Issue by RPGHacker** [RPGHacker/asar#24]
Macro labels can currently "leak" and affect sublabels outside of the macros. This shouldn't be the case; macro labels should be local to the macro and should disappear as soon as the macro scope is left.

Also add support for macro sublabels and macro +/- labels for consistency. Examples:

```
macro my_macro()

?MacroMain:
        ; ...

?.MacroSub:
        ;...

?-
        ;...
?+

endmacro
```

**Comment by RPGHacker** [comment]
Commit 6792965a37e388c58aee89c0c3af2c9f26563299 fixes a part of this problem, but not all of it, because it turns out that I missunderstood part of the original problem. I thought the original problem people had was that macro labels (as in `?Label`) leaked outside of macros. That's not the case, though. The actual problem they had was that normal labels (as in `Label`) "leak" outside of macros. In my eyes, this is the correct and expected behavior (I treat macros in Asar like macros in C and basically see them as "preprocessor" construct, even though they aren't implemented like that). It's only the way some people use macros that makes this a problem - they use them to both define and register subroutines, and they just call them inline in another routine. My commit makes macro variations of all supported label types possible, allowing you to use sub labels and +/- labels in macros without breaking anything outside of them by calling them, but using macro labels doesn't allow you to define a routine inside a macro and then call that routine outside of the macro. For this, you would have to use a main label, which would start a new sub label group and thus potentially break existing label hierarchies.

For this reason, I'm planning to introduce yet another type of label into Asar. I'm not sure on the specifics yet, but the new label type would have the property "defines a new main label without starting a new sub label group". I think that's the best workaround for this particular problem.

**Comment by RPGHacker** [comment]
Implemented in 7a09a3f45c5cf8dbeb8794aecf0ea6c371e19049
You can now prefix main label declarations (as well as sub label and the respective macro label declarations, but I don't think it has any practical use there) with a `#`, which will make Asar define the label without modifying existing label hierarchies. This can hopefully be used to fix the hacky routine macros in GPS.
